synapseclient/api/wiki_service.py

```diff
@@ -0,0 +1,456 @@
+async def put_wiki_version(
+    owner_id: str,
+    wiki_id: str,
+    wiki_version: int,
+    request: Dict[str, Any],
+    *,
+    synapse_client: Optional["Synapse"] = None,
+) -> Dict[str, Any]:
+    """Update a specific version of a wiki page.
+    <https://rest-docs.synapse.org/rest/PUT/entity/ownerId/wiki2/wikiId/version.html>
+
+    Arguments:
+        owner_id: The ID of the owner entity.
+        wiki_id: The ID of the wiki.
+        wiki_version: The version number to update.
+        request: The updated wiki page.
+        synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor.
+
+    Returns:
+        The updated wiki page version.
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+    return await client.rest_put_async(
+        uri=f"/entity/{owner_id}/wiki2/{wiki_id}/{wiki_version}",
+        body=json.dumps(request),
+    )
+
+
+async def delete_wiki_page(
+    owner_id: str,
+    wiki_id: str,
+    *,
+    synapse_client: Optional["Synapse"] = None,
+) -> None:
+    """Delete a wiki page.
+    <https://rest-docs.synapse.org/rest/DELETE/entity/ownerId/wiki2/wikiId.html>
+
+    Arguments:
+        owner_id: The ID of the owner entity.
+        wiki_id: The ID of the wiki.
+        synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor.
+    """
+    from synapseclient import Synapse
+
+    client = Synapse.get_client(synapse_client=synapse_client)
+    await client.rest_delete_async(
+        uri=f"/entity/{owner_id}/wiki2/{wiki_id}",
+    )
+
+
+async def get_wiki_header_tree(
+    owner_id: str,
+    offset: Optional[int] = 0,
+    limit: Optional[int] = 20,
+    *,
+    synapse_client: Optional["Synapse"] = None,
+) -> AsyncGenerator[Dict[str, Any], None]:
+    """Get the header tree (hierarchy) of wiki pages for an entity.
+    <https://rest-docs.synapse.org/rest/GET/entity/ownerId/wikiheadertree2.html>
+
+    Arguments:
+        owner_id: The ID of the owner entity.
+        offset: The index of the pagination offset. For a page size of 10, the first page would be at offset = 0,
+               and the second page would be at offset = 10. Default is 0.
+        limit: Limits the size of the page returned. For example, a page size of 10 requires limit = 10.
+               Limit must be 50 or less. Default is 20.
+        synapse_client: If not passed in and caching was not disabled by
+                `Synapse.allow_client_caching(False)` this will use the last created
+                instance from the Synapse class constructor.
+
+    Returns:
+        A generator over the wiki header tree for the entity. The tree contains the hierarchy of wiki pages
```

Comment:
For all of these functions where we are returning stuff like a dictionary could you linked to the rest API doc that shows the fields that are expected?